proposals/kroxylicious-operator-api-alternative.md

```diff
@@ -0,0 +1,654 @@
+
+```yaml
+apiVersion: proxy.kroxylicious.io/v1
+kind: ProxyIngress
+metadata:
+  name: myclusterip
+spec:
+  # reference to the proxy that will use this ingress.
+  proxyRef:
+    kind: Proxy  # if present must be Gateway, otherwise defaulted
+    group: proxy.kroxylicious.io # if present must be proxy.kroxylicious, otherwise defaulted=
+    name: myproxy  # name of proxy
+
+  # oneOf: clusterIP, loadBalancer, openShiftRoute, gateway
+  clusterIP:
+    protocol: TCP|TLS (ClusterIP supports both)
+    port: # optional, defaults to 9082
+  loadBalancer:
+    bootstrapAddressPattern: $(vitrtualClusterName).kafka.com 
+    brokerAddressPattern: broker-$(nodeId).$(vitrtualClusterName).kafka.com
+    port: # optional, defaults to 9082
+  openShiftRoute:
+    # ingress controller name (optional, if omit defaults to default  in the openshift-ingress-operator namespace).
+    resourceRef:
+      kind: IngressController  # if present must be IngressController, otherwise defaulted to IngressController
+      group: operator.openshift.io # if present must be operator.openshift.io, otherwise defaulted to operator.openshift.io
+      name: myingresscontroller
+      namespace: openshift-ingress-operator # namespace of the gateway, if omitted assumes namespace of this resource
+  gateway:
+    type: TLSRoute
+    resourceRef:
+      kind: Gateway  # if present must be Gateway, otherwise defaulted to Gateway
+      group: gateway.networking.k8s.io # if present must be gateway.networking.k8s.io, otherwise defaulted to gateway.networking.k8s.io
+      name: foo-gateway  # name of gateway resource to be used
+      namespace: namespace # namespace of the gateway, if omitted assumes namespace of this resource
+      sectionName: mytlspassthrough # refers to a TLS passthrough listener defined in the Gateway
+
+ # Optional
+ infrastructure:
+    # Controls the Admin influence the annotations/labels created on the ingress objects (Service, OpenShift Routes, TLSRoutes etc).
+    # This lets the Infrastructure admin  specify things like:
+    # https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/guide/service/annotations/#lb-type
+    annotations:
+       x: y
+    labels:
+       x: y
+status:
+   # describes the validity state of the ingress.  For instance:
+   # - for openShiftRoute the operator will verify that the ingress controller exists.
+   # - for gateway, it will verify that the gateway exists and the listener is defined as TLS passthrough.
+   conditions:
+   - status: x
+     type: y
+     reason: r 
+```      
+
+### VirtualCluster
+
+The VirtualCluster CR is the responsibilty of the the Developer.
+
+Declares a virtualcluster.
+
+A Virtualcluster is associated with exactly one Proxy.
+
+It enumerate the ingresses that are to be used by the virtualcluster.  It also supplies the virtual cluster ingress specific information
+such as the TLS certificates.
```

Comment:
TLS config and target cluster location feels more like an `Infrastructure admin` concern

Reply:
Yep, I can see that trying to separate out frontend and backend tls config would end up with an unwieldy amount of CRs required